modules/git/repo.go

```diff
@@ -304,6 +305,47 @@ func GetDivergingCommits(ctx context.Context, repoPath, baseBranch, targetBranch
 	return DivergeObject{ahead, behind}, nil
 }
 
+// CreateConfig creates a git config for the repo
+func CreateConfig(repoPath string) error {
+	return createConfig(repoPath)
+}
+
+// createConfig creates a git config for the repo
+func createConfig(repoPath string) (err error) {
+	gitConfigPath := filepath.Join(repoPath, "config")
+
+	reflogEnabled := setting.Git.Reflog.Enabled
+	reflogExpirationDays := setting.Git.Reflog.Expiration
+
+	configStr := strings.TrimSpace(`
+[core]
+	repositoryformatversion = 0
+	filemode = true
+	bare = true
+	ignorecase = true
+	precomposeunicode = true
+`)
+
+	if reflogEnabled {
+		configStr += "\n	logAllRefUpdates = true"
+		if reflogExpirationDays != 90 {
+			configStr += "\n[gc]"
+			configStr += fmt.Sprintf("\n	reflogExpire = %d", reflogExpirationDays)
+		}
+	}
+
+	configStr += "\n"
+
+	if err = util.Remove(gitConfigPath); err != nil && !os.IsNotExist(err) {
```

Comment:
I did not fully understand the purpose of deleting instead of just writing, but left it here since that's what the hooks task does as well.

Reply:
Argh is the hook task writing directly to .gitconfig?

You must not do this. It's fine to do that if you're managing .gitconfig by hand on a few repos but there will be concurrency issues if you're doing it on a server.

Use git config to write to .gitconfig instead.

Reply:
Hooks task simply writes hook scripts. I am not sure what issues specifically that locking would prevent, but I have a feeling that whatever it is, the hooks task would have the same issue.

At the very least, it seems like deleting hooks before re-writing them would cause concurrency issues if the hook was accessed during the moment in which it's deleted, I think you could at least eliminate that by just opening the file for writing instead of deleting first.

Reply:
ok so the hook tasks aren't writing to .git/config ... that's fine(ish) (There is a possible concurrency issue there but it's actually less of a problem.)

However, in this case you must use git config to adjust the config.

Reply:
I have no problem with that. I do have a question though, so assuming I only run a git config command, I probably will not want to fully sync the `gitconfigs` anymore and simply just change these two options (the reflog settings). Or should I try to be aware of all possible configuration settings since there may be some future use of these?

Alternatively I could even hoist these up to a more global gitconfig, so we don't always need to be modifying the individual repos.

Reply:
Yup, just setting the .gitconfig would work and we could that without a even a cron task - just set it on init.

Reply:
BTW I've just re-read what you're doing here and this code would have resulted in destroying mirrors!!

Reply:
Yep, sorry about that. First PR and all... I'm not sure what all the capabilities are yet so it will be a bit of a journey to avoid stepping on toes.

Reply:
OK, gitconfigs are no longer written, it's just stored in the global gitconfig now.